task_queue_consumer/task_queue_consumer.py

```diff
@@ -65,11 +66,59 @@ def sanitise_nmap_target(target_str):
             if is_valid_hostname(target):
                 sanitised.append(target)
             else:
-                raise ValueError(f"Target {target} was an invalid specification.")
+                raise ValueError(
+                    f"Target {target} was an invalid specification.")
 
     return " ".join(sanitised)
 
 
+def submit_ecs_task(event, host, message_id):
+    ssm_params = event["ssm_params"]
+    private_subnet = "true" == ssm_params[PRIVATE_SUBNETS]
+    network_configuration = {
+        "awsvpcConfiguration": {
+            "subnets": ssm_params[SUBNETS].split(","),
+            "securityGroups": [ssm_params[SECURITY_GROUP]],
+            "assignPublicIp": "DISABLED" if private_subnet else "ENABLED"
+        }
+    }
+    ecs_params = {
+        "cluster": ssm_params[CLUSTER],
+        "networkConfiguration": network_configuration,
+        "taskDefinition": ssm_params[IMAGE_ID],
+        "launchType": "FARGATE",
+        "overrides": {
+            "containerOverrides": [{
+                "name": task_name,
+                "environment": [
+                    # TODO The only bit of this file that isn't going to be the same for other
+                    # task queue executors, is this bit that maps the request body to some env vars
+                    # Extract the common code into a layer exported by the task-execution project
+                    {
+                        "name": "HOST_TO_SCAN",
```

Comment:
Should probably change this name to NMAP_TARGET_STRING too, based on my other comment. Will need to match that in the task script too.